backend/routes/follows.js

```diff
@@ -4,56 +4,113 @@ import { Stock } from "../models/stock.js";
 
 const router = express.Router();
 
+const checkSession = (req, res, next) => {
+    const user = req.session.user;
+    if (!user) {
+        res.status(401).json({ error: "Missing Session." });
+    } else {
+        next();
+    }
+};
+
+router.use(checkSession);
+
 router.get("/follows", async (req, res) => {
     const user = req.session.user;
 
-    const follows = await Follow.findAll({
-        where: { UserId: user.id },
-        include: [{ model: Stock }]
-    });
+    try {
+        const follows = await Follow.findAll({
+            where: { UserId: user.id },
+            include: [{ model: Stock }]
+        });
+
+        if (follows.length === 0) {
+            return res
```

Comment:
Nit: This works, and it is also pretty common. However I personally think it's more clear to separate the res and the return:
```
res
  .status(404)
  .json({ error: "This user doesn't follow any stocks." });
return;
```